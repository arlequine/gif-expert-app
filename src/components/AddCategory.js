import React, { useState } from 'react'
import PropTypes from 'prop-types';


const AddCategory = ({ setCategories }) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = e => {
    console.log(e.target.value)
    setInputValue(e.target.value)
  }
  const handleSubmit = e => {
    console.log('->handleSubmit', inputValue, '<--')
    e.preventDefault()
    console.log(inputValue)
    if (inputValue.trim().length > 2) {
      setCategories(categories => [inputValue, ...categories]);
      setInputValue('');
    }

  }
  return (
    <>
      <h2>Add Category</h2>
      <p>{inputValue}</p>
      <form onSubmit={handleSubmit}>
        <input
          type='text'
          value={inputValue}
          onChange={handleInputChange}
          placeholder='Category Name'
        />
      </form>
    </>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}

export default AddCategory;