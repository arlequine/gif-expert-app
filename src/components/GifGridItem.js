import React from 'react'
import PropTypes from 'prop-types';

const GifGridItem = (img) => {
  console.log(img)
  return (
    <div className="card animate__animated animate__fadeIn" >
      <img src={img.url} alt={img.title} />
      <p>{img.title}</p>
    </div>
  )
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
export default GifGridItem;