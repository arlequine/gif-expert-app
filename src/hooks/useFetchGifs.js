// el use en el archivo es nombre estandar para identificar el hook
import { useState, useEffect } from 'react';
import { getGifs } from '../helpers/GetGifs';
export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true
  })

  useEffect(() => {
    getGifs(category)
      //pasa por automatico el argumento y hace el set de las iamgenes en el state
      /*
      .then(setState)
      */
      .then(imgs => {
        setTimeout(() => {
          console.log(imgs);
          setState({
            data: imgs,
            loading: false
          })
        }, 3000);
      });
  }, [category])

  return state // { data: [],loading: false}
}
