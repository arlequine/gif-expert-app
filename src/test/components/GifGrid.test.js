import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')
// servira para fingir una llamada hacia ese archivo y controlar la informacion que devuelva

describe('Pruebas sobre el componente GifGrid', () => {
  const val = 'Hola Mundo'

  test('Comprobar que se muestre el componenten <GifGrid />', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    const wrapper = shallow(<GifGrid category={val} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
    const gifs = [
      {
        id: 1,
        title: 'One piece',
        url: 'http://www.gif.com/onepiece.gif',
      },
      {
        id: 2,
        title: 'Dragon ball',
        url: 'http://www.gif.com/dragonball.gif',
      },
      {
        id: 3,
        title: 'How i meet your mother',
        url: 'http://www.gif.com/howimeetyourmother.gif',
      }
    ]
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    const val = 'One Piece'
    const wrapper = shallow(<GifGrid category={val} />);

    // expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

  });


});
