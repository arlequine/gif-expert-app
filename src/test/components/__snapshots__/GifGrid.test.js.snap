// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente GifGrid Comprobar que se muestre el componenten <GifGrid /> 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Hola Mundo
  </h3>
  <p
    className=" animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
